library_test.py

```diff
@@ -155,21 +156,26 @@ async def main() -> None:
         print(f"You have to specify credentials for {args.email}")
         args.password = getpass.getpass("Password: ")
 
+    client_session = ClientSession()
+
     api = AmazonEchoApi(
+        client_session,
         args.country,
         args.email,
         args.password,
         login_data_stored,
-        args.save_raw_data,
     )
 
+    if args.save_raw_data:
+        api.save_raw_data()
```

Comment:
Need to check if this is async-safe

Reply:
`_save_to_file` is not async safe and should be running in the executor. And it may need an asyncio.Lock around it if it can be called multiple times

Reply:
This is a debug only call and is not used at all in HomeAssistant but only from library_test.py command line tools.